app/assets/javascripts/modules/all-content-finder.js

```diff
@@ -0,0 +1,53 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  // Enhances the all content finder with dynamic behaviour and analytics tracking
+  class AllContentFinder {
+    constructor (module) {
+      this.module = module
+
+      this.$form = module.querySelector('.js-all-content-finder-form')
+      this.$taxonomySelect = module.querySelector('.js-all-content-finder-taxonomy-select')
```

Comment:
If you're following the convention that dom elements need a $ prefix then this.module should have one too.